File: go-version/leetcode/LC1287/find_special_integer.go
package lc1287

func findSpecialInteger(arr []int) int {
	n := len(arr)
	num, cnt := arr[0], 0
	for i := 0; i < n; i++ {
		if num == arr[i] {
			cnt++
			if cnt*4 > n {
				return num
			}
		} else {
			num = arr[i]
			cnt = 1
		}
	}
	return -1
}

File: python-version/LC1287/find_special_integer.py
class Solution:
    def findSpecialInteger(self, arr: list[int]) -> int:
        n = len(arr)
        num, cnt = arr[0], 0
        for i in range(n):
            if arr[i] == num:
                cnt += 1
                if cnt * 4 > n:
                    return num
            else:
                num = arr[i]
                cnt = 1
        return -1